store/storage/db.go

```diff
@@ -0,0 +1,221 @@
+package storage
+
+import (
+	"fmt"
+	"sync"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	"cosmossdk.io/store/v2"
+)
+
+// TODO:
+//
+// - Define and implement pruning method(s)
+// - Import/snapshot method(s)???
+// - Should we make this an interface?
```

Comment:
> // - Should we make this an interface?
Basically yes.  Most of these methods are reimplemented on e.g. `rocksdb/` anyway. The only feature add here seems to be a hash map which the consumer could also handle.